front50-core/src/main/java/com/netflix/spinnaker/front50/model/plugin/PluginRepository.java

```diff
@@ -0,0 +1,96 @@
+package com.netflix.spinnaker.front50.model.plugin;
+
+import java.util.List;
+
+public interface PluginRepository<T> {
```

Comment:
Include interface-level docs. What is it, what's its used for, are there any implicit behaviors that implementations should adhere to, etc? Also useful to document what `T` is or isn't, since it's totally unbounded.